kalite/coachreports/static/js/coachreports/coach_reports/views.js

```diff
@@ -0,0 +1,343 @@
+
+        var model_id = model.get("exercise_id") || model.get("video_id") || model.get("content_id");
+        this.detail_view = new DetailPanelInlineRowView({
+            model: model,
+            contents_length: this.contents.length,
+            content_item: this.contents.find(function(item) {return item.get("id") === model_id;})
+        });
+        this.$el.after(this.detail_view.el);
+
+        this.trigger("detail_view", this.detail_view);
+
+    }
+
+});
+
+var TabularReportRowCellView = BaseView.extend({
+
+    tagName: 'td',
+
+    events: {
+        "click": "show_detail_view"
+    },
+
+    status_class: function() {
+        var status_class = "partial";
+        if (_.isEmpty(this.model.attributes)) {
+            status_class = "not-attempted";
+        } else if (this.model.get("complete")) {
+            status_class = "complete";
+        } else if (this.model.get("struggling")) {
+            status_class = "struggling";
+        }
+        return status_class;
+    },
+
+    className: function() {
+        return sprintf("status data %s", this.status_class());
+    },
+
+    title_attributes: {
+        "not-attempted": gettext("Not Attempted"),
+        "partial": gettext("Attempted"),
+        "complete": gettext("Complete"),
+        "struggling": gettext("Struggling")
+    },
+
+    attributes: function() {
+        return {title: this.title_attributes[this.status_class()]};
+    },
+
+    initialize: function() {
+        this.render();
+    },
+
+    render: function() {
+        if (this.model.has("streak_progress")) {
+            this.$el.html(this.model.get("streak_progress") + "%");
+        }
+    },
+
+    show_detail_view: function() {
+        if (_.isEmpty(this.model.attributes)) {
+            return false;
+        } else {
+            this.trigger("detail_view", this.model);
+        }
+    }
+});
+
+var FacilitySelectView = Backbone.View.extend({
+    template: HB.template('coach_nav/facility-select'),
+
+    initialize: function() {
+        _.bindAll(this);
+        this.facility_list = new FacilityCollection();
+        this.facility_list.fetch();
+        this.listenTo(this.facility_list, 'sync', this.render);
+        this.render();
+    },
+
+    render: function() {
+        this.$el.html(this.template({
+            facilities: this.facility_list.toJSON(),
+            selected: this.model.get("facility")
+        }));
+        if (!this.model.get("facility")) {
+            this.facility_changed();
+        }
+        return this;
+    },
+
+    events: {
+        "change": "facility_changed"
+    },
+
+    facility_changed: function() {
+        this.model.set("facility", this.$(":selected").val());
+    }
+});
+
+var GroupSelectView = Backbone.View.extend({
+    template: HB.template('coach_nav/group-select'),
+
+    initialize: function() {
+        this.group_list = new GroupCollection();
+        this.fetch_by_facility();
+        this.listenTo(this.group_list, 'sync', this.render);
```

Comment:
Same as above, you should register the listener before `fetch`ing, and what happens on the `error` event?